src/choice_map.jl

```diff
@@ -953,3 +953,24 @@ nested_view(c::ChoiceMap) = ChoiceMapNestedView(c)
 # aux data together.
 
 export nested_view
+
+#############################################
+# filtering choice maps based on selections #
+#############################################
+
+function get_selected(
+        choices::ChoiceMap, selection::Selection)
+    output = choicemap()
```

Comment:
I think we want this to return the same type as `choices`, not always a `DynamicChoiceMap`?

Reply:
I don't think so. User code does not and should not depend on the concrete type of a choice map. The exception is mutable choice maps (DynamicChoiceMaps) that the user constructs themselves. Otherwise choice maps all have the same API.

For example, some concrete choice map types are views into traces. The fact that the choice map is a view on a trace is a performance optimization --- there is nothing you can do with such a choice map that you cannot do with a DynamicChoiceMap.

Reply:
Ah ok.  Hmm... Does the API require that any trace type be able to interoperate with any kind of choice map?

Reply:
Yes.

Reply:
Ok, in that case SGTM.